### PR TITLE
Element-R: support for displaying QR codes during verification

### PR DESCRIPTION
**Pull request by richvdh**
Closes https://github.com/vector-im/crypto-internal/issues/124

---
Here's what your changelog entry will look like:

## ✨ Features
 * Element-R: support for displaying QR codes during verification ([\#3588](https://github.com/matrix-org/matrix-js-sdk/pull/3588)). Fixes vector-im/crypto-internal#124.

### CODE REVIEW

File: src/rust-crypto/verification.ts [hunk 1]
```diff
@@ -40,13 +41,16 @@ export class RustVerificationRequest
     extends TypedEventEmitter<VerificationRequestEvent, VerificationRequestEventHandlerMap>
     implements VerificationRequest
 {
+    /** a reëmitter which relays VerificationRequestEvent.Changed events emitted by the verifier */
```
Comment (t3chguy) [inline src/rust-crypto/verification.ts:44]: `ë`?

File: src/rust-crypto/verification.ts [hunk 2]
```diff
@@ -40,13 +41,16 @@ export class RustVerificationRequest
...
+    private readonly _reEmitter: TypedReEmitter<VerificationRequestEvent, VerificationRequestEventHandlerMap>;
```
Comment (t3chguy) [inline src/rust-crypto/verification.ts:45]: Why underscored?

**Review by t3chguy — APPROVED** [review]
Looks sane otherwise

File: src/rust-crypto/verification.ts (hunk 1 above)
Comment (richvdh) [reply src/rust-crypto/verification.ts:44]: https://www.merriam-webster.com/words-at-play/mary-norris-diaeresis

my use of it is tongue-in-cheek. Will replace if you find it distracting/confusing.

File: src/rust-crypto/verification.ts (hunk 2 above)
Comment (t3chguy) [reply src/rust-crypto/verification.ts:45]: Our code style is explicit here

> Private members are private and not prefixed unless required for naming conflicts.

https://github.com/vector-im/element-web/blob/develop/code_style.md#typescript--javascript

Comment (richvdh) [reply src/rust-crypto/verification.ts:45]: yes, my question is do you want me to fix the other properties here or in a separate PR

Comment (t3chguy) [reply src/rust-crypto/verification.ts:45]: Happy for it to be here, approval stands

Comment (richvdh) [reply src/rust-crypto/verification.ts:45]: turns out the other properties *should* have underscores because there are getter methods which conflict

**Pull request merged**